Sources/Layout/MessageSizeCalculator.swift

```diff
@@ -134,7 +134,7 @@ open class MessageSizeCalculator: CellSizeCalculator {
         return position
     }
 
-    public func avatarSize(for message: MessageType) -> CGSize {
+    open func avatarSize(for message: MessageType) -> CGSize {
```

Comment:
We originally made these non-overrideable because allowing dynamic layouts can lead to performance and cache invalidation issues. I think all of these `public` properties can now become `open` tbh. It should be proceed at your own risk at this point

Reply:
👍I like the idea of having a couple of good optimised defaults for the basic types of cells, and being able to deeply customise others. Thanks for explaining!

Reply:
I still think all functions above `// MARK: - Helpers` should be made open. If you want to make that change, I'll accept the PR

Reply:
sure thing!

Reply:
cc @zhongwuzw @nathantannar4, I just wanted to ping you one this one. 

We've essentially come full circle. Originally, we made this public to prevent the user from creating a dynamic layout because it broke our internal cache and had perf implications. But after @zhongwuzw's awesome optimizations 🤗 we ditched the cache (and I don't intend to add it back because the logic for cache invalidation is way too client specific)

Thus, I see no reason for preventing users from doing this any longer

Reply:
Emm, if we decide that would not do any cache, maybe we can move to `open`.

Reply:
Yeah @zhongwuzw, we're not going to add a cache. The client should be responsible for this. It's too client specific. How should we invalidate the cache if a section is reloaded? Other sections could depend on changes in that section (ie. stacking avatars)